### PR TITLE
Adds a handful of Ninja Hacking MODule interactions of varying usefulness

### PR DESCRIPTION
**Pull request by Rhials**
## About The Pull Request

Adds a few new interactions with the Ninja's ~~right click multipurpose trolling tool~~ Hacking MOD Module. These new effects are not tied to objectives and are geared towards expanding the ninja's access, disabling equipment, and giving them more ways to punk the crew.

### **Useful additions**
Ninjas can now hack open **windoors** and **elevator control panels**. Both trigger emag_act() when hacked, opening in the case of the windoor, and disabling the access restrictions _(and also maybe the safeties)_ in the case of the elevator controls.

**Buttons** can also be emagged by the hacking modules, which removes their access restrictions.

Hacking a **camera** will now EMP it, disabling it for about 90 seconds. This can especially useful when trying to complete objectives, and works better than smashing the cameras with your sword or lugging around tools.

**Firelocks** can be right-click opened now too, thanks to the hacking MODule.

**Energy guns** of all variety, useless to a ninja since they can't use ranged weapons, can now be drained and used to charge your suit. This takes a brief do_after to complete, so pulling it off mid-combat may be as risky as it is stylish. 

### **Being a nuisance**

**Vendors** can be hacked, expending some charge to trigger the "throw" wire, making it launch inventory at anyone who passes by.

You can now hack **simplebots**, expending some charge from your suit to overload and detonate them. It's faster than tipping, and far more tragic. Sentient simplebots should take care when a ninja is around.

### **Sabotage opportunities**

The **recycler** can now be hacked. This takes 30 seconds and notifies the AI, just like the communications console hack. Completing the hack will emag it. That's it.

Hacking the **tram control console** will trigger an extended Tram Malfunction event, and can be repeated after the event is done. This can only be done to the "main" tram of the map, which I forsee will be an absolute nightmare to complete on highpop tramstation.

Neither of these, presently, contribute towards any objectives. They can be useful for diverting attention, but I see them more as ways for an overachieving ninja to flex or continue the chaos after their objectives are complete.

### **OH ALSO**

Hacking open doors costs energy. This really shouldn't be an issue just remember to recharge sometimes.

The charge drains and do_after lengths for all of these were chosen on vibes. In all honesty I think the drainage values don't _really_ matter, due to how easy recharging is, but I had a hard time settling on how long some of these hack do_afters should take (even if I know they probably won't matter either).

## Why It's Good For The Game

Being able to hack open airlocks but not windoors always irked me. I figured that they would be openable like any other door, and that losing their status as a "-1 dash charge gate" wouldn't be a big enough balance change to spark arguments. The same philosophy extends to buttons and elevator controls.

Sapping power from eguns expands on the list of sources energy can be siphoned from. You can also use it to disarm opponents (like the badass ninja you are), take emergency charge from a recently-gored officer's disabler, or dunk on security by draining their entire armory.

Hacking simplebots, vendors, and by extension elevator lifts (since that also disables the safeties!) give opportunities for minor sabotage. Not meant to be super disruptive, just a bit silly and annoying for the crew.

The recycler/tram hacking in particular are meant to be bonus goals, only sought out by the ballsiest (or more likely boredest) of ninjas. 

I see all of these additions as expanding upon the current abilities of the ninja (not really making them much more powerful, just expanding their flexibility), or providing more interactions to have fun (and antagonize the crew) with when not mainlining their objectives.
## Changelog
:cl: Rhials
add: Ninjas can now temporarily disable cameras with the Ninja MOD right-click hacking ability.
add: Ninjas can emag windoors, elevator controls, and buttons with their hacking ability.
add: Ninjas can drain the power from energy weaponry, adding the charge to their MODsuit.
add: Ninjas can now hack simplebots, overloading and detonating them after a brief delay.
add: Ninjas can now hack vendors, causing them to eject their inventory at people.
add: Ninjas can now hack the recycler, which notifies the AI and emags it once complete.
add: Ninjas can now trigger an extended tram malfunction by hacking the tram control console.
add: Ninjas can now hack open firelocks (temporarily) with right-click.
balance: Hacking open doors with the Ninja Hacking MODule will subtract a paltry amount of energy from your suit.
/:cl:

### CODE REVIEW

File: code/modules/antagonists/ninja/ninjaDrainAct.dm [hunk 1]
```diff
@@ -322,3 +322,135 @@
...
+	addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(explosion), src, 0, 1, 2, 3), 2.5 SECONDS)
+	return COMPONENT_CANCEL_ATTACK_CHAIN
+
+/mob/living/simple_animal/bot/medbot/ninjadrain_act(mob/living/carbon/human/ninja, obj/item/mod/module/hacker/hacking_module)
+	var/static/list/death_cry = list(
+		MEDIBOT_VOICED_NO_SAD,
+		MEDIBOT_VOICED_OH_FUCK,
+	)
+	speak(pick(death_cry))
+	. = ..()
```
Comment (Fikou) [inline code/modules/antagonists/ninja/ninjaDrainAct.dm:359]:
```suggestion
	return ..()
```

**Comment by Loafin34** [comment]
Why does emagging the recycler of all things require a 30 second do-after and a big old announcement about it? in order for a ninja to successfully use it, they would have to use tools/make a saboteur borg and then bring people directly into the portal/disposals chute for their gimmick.

**Comment by nukularpower** [comment]
really sick of seeing AI's immediately cutting power when ninjas start hacking so love this

**Comment by Rhials** [comment]
> Why does emagging the recycler of all things require a 30 second do-after and a big old announcement about it?

Because I thought it would be funny, and if it were easy it would happen every time

**Pull request merged**